CONTRIBUTING.md

```diff
@@ -1,60 +1,145 @@
+Requirements:
+* [Terraform](https://www.terraform.io/downloads.html) >= 1.0.0
+* [Go](https://golang.org/doc/install) >= 1.25
+* Docker (for acceptance tests)
 
-```bash
-make docker-testacc
-```
+## Development Workflow
 
-Run a single test with terraform debug enabled:
-```bash
-env TF_LOG=DEBUG make docker-testacc TESTARGS='-run ^TestAccResourceDataStreamLifecycle$$'
-```
+* Create a new branch for your changes.
+* Make your changes. See [Useful Commands](#useful-commands) and [Debugging](#debugging).
+  * GitHub Copilot can be used to help with these changes (See [Using Copilot](#using-copilot))
+* Validate your changes
+  * Run unit and acceptance tests (See [Running Acceptance Tests](#running-acceptance-tests)).
+  * Run `make fmt`, `make lint`.
+  * All checks also run automatically on every PR.
+* Add a changelog entry in `CHANGELOG.md` under the `Unreleased` section. This will be included in the release notes of the next release.
+* Submit your PR for review.
 
-A way to forward debug logs to a file:
-```bash
-env TF_ACC_LOG_PATH=/tmp/tf.log TF_ACC_LOG=DEBUG TF_LOG=DEBUG make docker-testacc
-```
+When creating new resources:
+* Use the [Plugin Framework](https://developer.hashicorp.com/terraform/plugin/framework/getting-started/code-walkthrough) for new resources.
+  * Use an existing resource (e.g. `internal/elasticsearch/security/system_user`) as a template.
+  * Some resources use the deprecated Terraform SDK, so only resources using the new Terraform Framework should be used as reference.
+* Use the generated API clients to interact with the Kibana APIs. (See [Working with Generated API Clients](#working-with-generated-api-clients)
+* Add documentation and examples for the resource. Update the generated docs with `make docs-generate`.
+* Write unit and acceptance tests.
 
 
-## Update documentation
+## Using Copilot
 
-Update documentation templates in `./templates` directory and re-generate docs via:
-```bash
-make docs-generate
-```
+GitHub Copilot can speed up development, but you’re responsible for correctness:
+* Create an issue describing the desired change and acceptance criteria.
+* Assign the issue to Copilot and iterate with prompts.
+* Review outputs carefully; add tests and adjust as needed.
+* Example issue: https://github.com/elastic/terraform-provider-elasticstack/issues/1219
 
-## Update `./CHANGELOG.md`
+### Useful Commands
 
-List of previous commits is a good example of what should be included in the changelog.
+* `make build`: Build the provider.
+* `make install`: Install the provider.
+* `make fmt`: Format the code.
+* `make lint`: Lint the code.
+* `make test`: Run unit tests.
+* `make docs-generate`: Generate documentation.
 
+#### Running Acceptance Tests
 
-## Pull request
+Acceptance tests spin up Elasticsearch, Kibana, and Fleet with Docker and run tests in a Go container.
 
-Format the code before pushing:
+Quick start (default stack version from Makefile):
 ```bash
-make fmt
+make docker-testacc
```

Comment:
This path is can be quite slow, it both spins up the Elastic Stack docker containers (Yay!), but also runs the actual provider tests inside a golang docker container.

Repeated testing (i.e what we'd expect in a dev workflow) is much faster if we re-use a single running Stack environment but also run the tests directly on the dev machine. 

 ```suggestion
make docker-fleet
make testacc
```

Reply:
Sounds good, I updated the docs to use this faster workflow.